src/core/cache/value.ts

```diff
@@ -0,0 +1,19 @@
+import { CacheKeyFragment, getCache } from '.';
+
+export default <TKey extends CacheKeyFragment[]>() =>
+    <TEntry>(fn: (...a: TKey) => TEntry) => {
+        const cache = new Map<CacheKeyFragment, any>();
+
+        function get(...key: TKey) {
+            const lastKey = key.slice(-1)[0];
+
+            const nestedCache = getCache(cache, ...key);
+
+            return (
+                nestedCache.get(lastKey) ||
```

Comment:
What if `fn(...key)` is falsy? That can't happen in `memoizer` (returns a function) or `getCache` (returns a `Map`) but here it could, right?

Reply:
Ah, I guess the only way you're currently using the value cache is to store functions so not a problem in practice - still, in case it ever finds a use with primitives it might be nice to protect against that case.

Reply:
Actually this is a good point -- I use this to store the result of the ast's evaluation for the dropdowns -- a `false` eval will reevaluate -- always.

Reply:
> I guess the only way you're currently using the value cache is to store functions so not a problem in practice

No. In the value cache I store the actual value returned by the function, not the decorated function as in the memoizer cache.

Reply:
So the memoizer cache is fine but the value cache is incorrect. 🥇